src/js/processor.js

```diff
@@ -0,0 +1,285 @@
+
+  if (kbLang == 'eng') {
+    kb.childNodes.forEach((btn, i) => {
+      btn.innerHTML = constants.kbRusLower[i];
+    });
+    localStorage.setItem('kbLang', 'rus');
+
+  } else if (kbLang == 'rus') {
+    kb.childNodes.forEach((btn, i) => {
+      btn.innerHTML = constants.kbEngLower[i];
+    });
+    localStorage.setItem('kbLang', 'eng');
+  }
+}
+
+// handle pressed key event
+export function physicalClickHandler() {
+    let tArea = document.querySelector('.textarea-block');
+
+    document.addEventListener('keydown', function(event) {
+      highlightKey(event.code, event.key);
+
+      document.addEventListener('keyup', function (event) {
+        darkenKey(event.code, event.key);
+      });
+
+      if (event.location == 3) {
+        return;
+      }
+
+      // if alt & shift are pressed together
+      if (event.altKey && event.shiftKey) {
+        event.preventDefault();
+        changeLang();
+        return;
+      }
+
+      if (event.code == 'ShiftLeft' || event.code == 'ShiftRight') {
+        return;
+      }
+
+      if (event.ctrlKey) {
+        return;
+      }
+
+      if (event.altKey) {
+        event.preventDefault();
+        return;
+      }
+
+      if (event.code == 'Tab') {
+        event.preventDefault();
+        tArea.innerHTML += '    ';
+        return;
+      }
+
+      if (event.code == 'CapsLock') {
+        changeRegister();
+        return;
+      }
+
+      if (event.code == 'Backspace') {
+        let tAreaText = document.querySelector('.textarea-block').innerHTML;
+        tAreaText = tAreaText.substring(0, tAreaText.length - 1);
+        tArea.innerHTML = tAreaText;
+        return;
+      }
+
+      if (event.code == 'Delete') {
+        let tAreaText = document.querySelector('.textarea-block').innerHTML;
+        tAreaText = tAreaText.substring(1, tAreaText.length);
+        tArea.innerHTML = tAreaText;
+        return;
+      }
+
+      if (event.code == 'Enter') {
+        tArea.innerHTML += '\n';
+        return;
+      }
+
+      if (event.metaKey) {
+        event.preventDefault();
+        return;
+      }
+
+      // if arrow key is pressed
+      if (event.code.match(constants.arrowRegExp)) {
+        if (event.code == 'ArrowUp') {
+          tArea.innerHTML += '↑';
+          return;
+        }
+
+        if (event.code == 'ArrowDown') {
+          tArea.innerHTML += '↓';
+          return;
+        }
+
+        if (event.code == 'ArrowLeft') {
+          tArea.innerHTML += '←';
+          return;
+        }
+
+        if (event.code == 'ArrowRight') {
+          tArea.innerHTML += '→';
+          return;
+        }
+        return;
+      }
+
+      // if any other key is pressed
+      tArea.innerHTML += event.key;
+    })
+}
+
+export const virtualClickHandler = function () {
+  let kb = document.querySelector('.keyboard-block');
+  let tArea = document.querySelector('.textarea-block');
+
+  kb.onclick = function (event) {
+    let target = event.target;
```

Comment:
here is a very good place to use Destructuring assignment.

```let { target } = event;```

Reply:
Done